mono/mini/mini-arm64.c

```diff
@@ -3360,19 +3387,22 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			arm_subx (code, ARMREG_IP1, ARMREG_IP1, ARMREG_IP0);
 			arm_movspx (code, ARMREG_SP, ARMREG_IP1);
 
-			/* Init */
-			/* ip1 = pointer, ip0 = end */
+			/* Init - (from high to low for Windows stack probes */
+			/* ip1 = start, ip0 = pointer */
 			arm_addx (code, ARMREG_IP0, ARMREG_IP1, ARMREG_IP0);
 			buf [0] = code;
 			arm_cmpx (code, ARMREG_IP1, ARMREG_IP0);
 			buf [1] = code;
 			arm_bcc (code, ARMCOND_EQ, 0);
-			arm_stpx (code, ARMREG_RZR, ARMREG_RZR, ARMREG_IP1, 0);
-			arm_addx_imm (code, ARMREG_IP1, ARMREG_IP1, 16);
+			arm_subx_imm (code, ARMREG_IP0, ARMREG_IP0, 16);
+			arm_stpx (code, ARMREG_RZR, ARMREG_RZR, ARMREG_IP0, 0);
 			arm_b (code, buf [0]);
 			arm_patch_rel (buf [1], code, MONO_R_ARM64_BCC);
 
 			arm_movspx (code, dreg, ARMREG_SP);
+
+			code = emit_prolog_setup_sp_win64 (cfg, code, cfg->param_area);
```

Comment:
Wondering why there is not an equivalent amd64 change for this? (and the OP_LOCALLOC_IMM case)

Reply:
The amd64 code is very different - it calls `mono_emit_stack_alloc` which does appear to handle the `param_area`.  So I don't believe that needs to be changed.